documentation/docs/tutorials/single-topic-multi-tenant.mdx

```diff
@@ -0,0 +1,341 @@
+
+## Running the Example
+
+### Using the provided script
+
+```bash
+cd docker
+./run-example-with-single-topic-multi-tenant.sh
+```
+
+### Manual setup
+
+Start the services:
+
+```bash
+docker compose -f docker-compose-single-topic-multi-tenant.yml up -d
+```
+
+Verify services are running:
+
+```bash
+curl http://localhost:7002/healthcheck   # OPAL Server
+curl http://localhost:8181/health        # OPA
+curl http://localhost:8090/acl/tenant1   # External Data Provider
+```
+
+## Adding Tenants
+
+### Add first tenant
+
+```bash
+curl -X POST http://localhost:7002/data/config \
+  -H "Content-Type: application/json" \
+  -d '{
+    "entries": [{
+      "url": "http://example_external_data_provider:80/acl/tenant1",
+      "topics": ["tenant_data"],
+      "dst_path": "/acl/tenant1"
+    }],
+    "reason": "Add tenant1 data source"
+  }'
+```
+
+### Add additional tenants
+
+```bash
+curl -X POST http://localhost:7002/data/config \
+  -H "Content-Type: application/json" \
+  -d '{
+    "entries": [{
+      "url": "http://example_external_data_provider:80/acl/tenant2",
+      "topics": ["tenant_data"],
+      "dst_path": "/acl/tenant2"
+    }],
+    "reason": "Add tenant2 data source"
+  }'
+```
+
+Note: No service restart is required when adding new tenants.
+
+## Verification
+
+### Check data isolation
+
+```bash
+# Tenant 1 data
+curl -s http://localhost:8181/v1/data/acl/tenant1 | jq .
+
+# Tenant 2 data
+curl -s http://localhost:8181/v1/data/acl/tenant2 | jq .
+
+# All tenant data
+curl -s http://localhost:8181/v1/data/acl | jq .
+```
+
+### Monitor OPAL logs
+
+OPAL Server logs:
+```bash
+docker compose -f docker-compose-single-topic-multi-tenant.yml logs -f opal_server
+```
+
+OPAL Client logs:
+```bash
+docker compose -f docker-compose-single-topic-multi-tenant.yml logs -f opal_client
+```
+
+Look for these log patterns:
+
+**Server side:**
+```
+Publishing data update to topics: {'tenant_data'}
+Broadcasting incoming event: {'topic': 'tenant_data'}
+```
+
+**Client side:**
+```
+Received notification of event: tenant_data
+Fetching data from url: http://example_external_data_provider:80/acl/tenant1
+Saving fetched data to policy-store: destination path='/acl/tenant1'
+```
+
+## Data Updates
+
+### Update existing tenant data
+
+```bash
+curl -X POST http://localhost:7002/data/config \
+  -H "Content-Type: application/json" \
+  -d '{
+    "entries": [{
+      "url": "http://example_external_data_provider:80/acl/tenant1",
+      "topics": ["tenant_data"],
+      "dst_path": "/acl/tenant1"
+    }],
+    "reason": "Refresh tenant1 data"
+  }'
+```
+
+## Testing Authorization
+
+### Test tenant-specific policies
+
+```bash
+curl -X POST http://localhost:8181/v1/data/policies/rbac/allow \
+  -H "Content-Type: application/json" \
+  -d '{
+    "input": {
+      "user": "alice",
+      "action": "read",
+      "resource": "document1",
+      "tenant_id": "tenant1"
+    }
+  }' | jq .
+```
+
+## Production Considerations
```

Comment:
This is slight duplicate of other articles on the OPAL docs, but its good to place it here too